ts/features/bonus/bpd/screens/iban/insertion/IbanInsertionComponent.tsx

```diff
@@ -73,23 +75,28 @@ export const IbanInsertionComponent: React.FunctionComponent<Props> = props => {
             </Body>
             <View spacer={true} large={true} />
             <H5>{ibanDescription}</H5>
-            <Item error={isInvalidIban}>
-              <Input
-                value={iban}
-                style={IBANInputStyle}
-                autoCapitalize={"characters"}
-                maxLength={IbanMaxLength}
-                onChangeText={text => {
+            <LabelledItem
+              type="text"
+              isValid={fromNullable(iban ? iban : undefined).fold(
```

Comment:
You've made a first step towards a more functional paradigm, this is great 🎉 🥳 
Now you might want to take full advantage of it by changing the way `iban` is defined rather than transforming it here 😉